engine/src/main/java/org/terasology/rendering/dag/stateChanges/BindFBO.java

```diff
@@ -19,34 +19,40 @@
 import org.terasology.rendering.opengl.BaseFBOsManager;
 import org.terasology.rendering.opengl.FBOManagerSubscriber;
 import com.google.common.base.Objects;
-import org.terasology.rendering.dag.RenderPipelineTask;
 import org.terasology.rendering.dag.StateChange;
 
 import static org.lwjgl.opengl.EXTFramebufferObject.GL_FRAMEBUFFER_EXT;
 import static org.lwjgl.opengl.EXTFramebufferObject.glBindFramebufferEXT;
 
 /**
- * TODO: Add javadocs
+ * Binds the given FBO to the GL_FRAMEBUFFER target, making it the read framebuffer target as well as
+ * the draw framebuffer target.
+ *
+ * This is useful for operations such as off-screen rendering.
  */
 public final class BindFBO implements FBOManagerSubscriber, StateChange {
-
     private static final Integer DEFAULT_FRAME_BUFFER_ID = 0;
     // TODO: add necessary checks for ensuring generating FBO with the name "display" is not possible.
     private static final ResourceUrn DEFAULT_FRAME_BUFFER_URN = new ResourceUrn("engine:display");
 
-    private static BindFBO defaultInstance = new BindFBO(DEFAULT_FRAME_BUFFER_URN);
+    private static BindFBO defaultInstance = new BindFBO();
 
-    private BindFBOTask task;
-    private BaseFBOsManager fboManager;
     private ResourceUrn fboName;
+    private BaseFBOsManager fboManager;
+    private int fboId;
```

Comment:
The change I made was to make the structure more consistent. Agreed, it should have been in a separate commit.. but it will happen nonetheless. Do you mind if I skip the reverting this one time, and make sure to do it in a separate commit next time onwards?